src/commands/introspect.ts

```diff
@@ -1,125 +1,183 @@
-import type { z } from "zod";
+import { z } from "zod";
 import type { CommandConfig } from "../cli/command";
 import { baseCommandArgsSchema, confirmAction } from "../cli/command";
 import { cliConsole } from "../cli/console";
 import { createConfigurator } from "../core/configurator";
 import { createBackup, fileExists } from "../lib/utils/file";
-
-export const introspectCommandSchema = baseCommandArgsSchema;
+import { 
+  parseSelectiveOptions, 
+  getSelectiveOptionsSummary 
+} from "../lib/utils/selective-options";
+
+export const introspectCommandSchema = baseCommandArgsSchema.extend({
+  dryRun: z.boolean().default(false).describe("Preview changes without applying them"),
+  noBackup: z.boolean().default(false).describe("Skip backup creation"),
```

Comment:
```suggestion
  isBackedUp: z.boolean().default(true).describe("Represents whether Configurator should create a backup schema"),
```

I suggest we avoid negative flags, they are harder to read